File: components/services/zend/src/app/services/test_service.php
<?php 
	class TestService
	{
		/**
		 * this is a test service
		 *
		 * @Service(request=app.test.message.request,response=app.test.message.response,version=1.0)
		 */
		function myServiceMethod ($request, $response)
		{
            $response->data['message'] = 'I received from you: ' . $request->data['message'];
            $response->data['success'] = 'true';
		}
	}
?>

File: components/services/ruby/src/app/services/test_service.rb

class TestService < Appcelerator::Service
  
  Service 'app.test.message.request', :testMessage, 'app.test.message.response'
  
  def testMessage
      logger.info "received message: #{params.inspect}"
      msg = params["message"]
      {"message"=>"I received from you: #{msg}","success"=>true}
  end
  
  def another_test
    #
    # this method can be invoked using the pattern:
    # 
    # test.another.test.request 
    # test.another.test.response
    #
    # where the pattern is <Class_excluding_Service>.<method_underscored_with_dots> 
    #
    # and you don't need to import with Service method above
    #
    logger.info "received message: #{params.inspect}"
    msg = params["message"]
    {"message"=>"I received from you: #{msg}","success"=>true}
  end
  
end

